core/src/main/kotlin/com/strumenta/kolasu/serialization/JsonGenerator.kt

```diff
@@ -187,8 +187,9 @@ class JsonGenerator {
         withDestinationIds: IdentityHashMap<Node, String>? = null
     ):
         JsonElement {
+        val nodeType = node.nodeType
         val jsonObject = jsonObject(
-            JSON_TYPE_KEY to if (shortClassNames) node.javaClass.simpleName else node.javaClass.canonicalName,
+            JSON_TYPE_KEY to if (shortClassNames) nodeType.substring(nodeType.indexOf('.') + 1) else nodeType,
```

Comment:
I think you want to find the last index of '.'. Alternatively I would use `nodeType.split('.').last()`

Reply:
Right, good catch!